pkg/trojan/message.go

```diff
@@ -163,32 +164,47 @@ func (m *Message) toChunk(targets Targets, span []byte) (swarm.Chunk, error) {
 		return nil, err
 	}
 
-	// hash chunk fields with different nonces until an acceptable one is found
-	for start := time.Now(); ; {
-		s := append(append(span, nonce...), b...) // serialize chunk fields
-		hash1, err := hashBytes(s)
-		if err != nil {
-			return nil, err
-		}
+	doneC := make(chan bool, 1)
```

Comment:
I think that there is no need to have two channels now. `doneC` can be removed as sending a nil to `errC` would act the same, just to check it on the receiving end if it is nil. This would also, it would not require a buffered channel.

Reply:
Alternatively, the mining code could have been written like this:
```go
t := time.After(MinerTimeout * time.Second)
for {
   select {
      case <-t:
          return nil, t.Err()
      default:
   }
   ...mine stuff here...
}
```
Personally I find this simpler and easier to read. But up to you